GVFS/GVFS/CommandLine/DehydrateVerb.cs

```diff
@@ -145,31 +164,139 @@ protected override void Execute(GVFSEnlistment enlistment)
+                                this.WriteMessage(tracer, $"Make sure there aren't any applications accessing the folder and try again.");
+                            }
+                            else
+                            {
+                                foldersToDehydrate.Add(folder);
+                            }
+                        }
+                        else
+                        {
+                            this.WriteMessage(tracer, $"{folder} did not exist to dehydrate.");
+                        }
+                    }
+
+                    return true;
+                },
+                "Backing up folders"))
+            {
+                this.ReportErrorAndExit(tracer, "Dehydrate for folders failed.");
+            }
+
+            this.Mount(tracer);
+
+            this.SendDehydrateMessage(tracer, enlistment, backupRoot, foldersToDehydrate.ToArray());
+        }
 
-                        this.Output.WriteLine();
-                        this.WriteMessage(tracer, "The repo was successfully dehydrated and remounted");
+        private void SendDehydrateMessage(ITracer tracer, GVFSEnlistment enlistment, string backupRoot, string[] folders)
+        {
+            NamedPipeMessages.DehydrateFolders.Response response = null;
+
+            try
+            {
+                using (NamedPipeClient pipeClient = new NamedPipeClient(enlistment.NamedPipeName))
+                {
+                    if (!pipeClient.Connect())
+                    {
+                        this.ReportErrorAndExit("Unable to connect to GVFS.  Try running 'gvfs mount'");
                     }
+
+                    NamedPipeMessages.DehydrateFolders.Request request = new NamedPipeMessages.DehydrateFolders.Request(string.Join(";", folders));
+                    pipeClient.SendRequest(request.CreateMessage());
+                    response = NamedPipeMessages.DehydrateFolders.Response.FromMessage(NamedPipeMessages.Message.FromString(pipeClient.ReadRawResponse()));
                 }
-                else
+            }
+            catch (BrokenPipeException e)
+            {
+                this.ReportErrorAndExit("Unable to communicate with GVFS: " + e.ToString());
+            }
+
+            if (response != null)
+            {
+                foreach (string folder in response.SuccessfulFolders)
                 {
-                    this.Output.WriteLine();
-                    this.WriteMessage(tracer, "ERROR: Backup failed. We will attempt to mount, but you may need to reclone if that fails");
+                    this.WriteMessage(tracer, $"{folder} folder successfully dehydrated.");
+                }
 
+                foreach (string folder in response.FailedFolders)
+                {
+                    this.WriteMessage(tracer, $"{folder} folder failed to dehydrate. You may need to clean the working directory and retry the dehydrate.");

```

Comment:
> may need to clean the working directory

What does "clean the working directory" mean in this context?  

Are there more explicit instructions we can provide (e.g. unmount your repo and delete the failed folder, run `git clean` for the folder, etc.)?